### PR TITLE
Fix MethodEnter support for Perf Agent

### PR DESCRIPTION
**Pull request by mpirvu**
In order to enable method tracing we need to set the
`can_generate_method_entry_events` capability. For some
reason this capability can be enabled only during the
OnLoad phase. Therefore, the agent must be loaded from
the very beginning, and late attach will not do.

This commit also adds the ability to print the stack trace
up to the specified depth.

Examples of commands:
```
  {
    "functionality": "methodEntryEvents",
    "command": "start",
    "delay": 1,
    "stackTraceDepth": 5
  },
  {
    "functionality": "methodEntryEvents",
    "command": "stop",
    "delay": 5
  }
```

Signed-off-by: Marius Pirvu <mpirvu@ca.ibm.com>

### CODE REVIEW

**Comment by mpirvu** [comment]
@dsouzai Could you please review this PR? Thanks

File: perf-tool/src/agentOptions.cpp [hunk 1]
```diff
@@ -175,34 +180,23 @@ void modifyObjectAllocEvents(const std::string& function,const std::string& comm
...
-    }
-    else if (!command.compare("stop"))
-    {
-        setMonitorStackTrace(false);
-    }
-    else
-    {
-        invalidCommand(function, command);
-    }
-}
```
Comment (dsouzai) [inline perf-tool/src/agentOptions.cpp:193]: This seems to be unrelated to the method enter/exit events; is this deleting the ability for the user to send `monitorStackTrace`?

File: perf-tool/src/methodEntry.cpp [hunk 2]
```diff
@@ -33,65 +33,76 @@
...
             err = jvmtiEnv->Deallocate((unsigned char*)table_ptr);
             check_jvmti_error(jvmtiEnv, err, "Unable to deallocate table_ptr.\n");
         }
-    
-        std::string s = j.dump();
-        sendToServer(s);
+*/
+        sendToServer(j.dump());
     }
-    mEntrySampleCount = atomic_fetch_add(&mEntrySampleCount, 1);
```
Comment (dsouzai) [inline perf-tool/src/methodEntry.cpp:96]: Don't we need this? Otherwise `mEntrySampleCount` will never get incremented.

**Review by dsouzai — CHANGES_REQUESTED** [review]
Should also remove the `void setMethodEntrySampleRate(int rate);` declaration in `methodEntry.hpp`.

File: perf-tool/src/agentOptions.cpp (hunk 1 above)
Comment (mpirvu) [reply perf-tool/src/agentOptions.cpp:193]: Yes. The stacktrace option is sent with the "start" command.
The exiting code actually did nothing

File: perf-tool/src/methodEntry.cpp (hunk 2 above)
Comment (mpirvu) [reply perf-tool/src/methodEntry.cpp:96]: There is another increment on line 45. This one was a duplicate.

**Comment by dsouzai** [comment]
@mpirvu just waiting for the declaration of `setMethodEntrySampleRate` to be removed from `methodEntry.hpp` before I merge.

**Comment by mpirvu** [comment]
@dsouzai I deleted the stale declaration of `setMethodEntrySampleRate` and corrected a couple of places where `sendToServer` is called to add a second parameter (event type), which is needed after #37 was merged.

**Review by dsouzai — APPROVED** [review]
